libs/diagnostics/src/performance_handler.cpp

```diff
@@ -20,15 +20,14 @@ static logging::Logger DIAG_LOGGER = logging::getLogger("concord.diag.perf");
 namespace concord::diagnostics {
 
 void Recorder::record(int64_t val) {
-  if (!hdr_interval_recorder_record_value(&(recorder), val)) {
-    LOG_WARN(DIAG_LOGGER, "Failed to record value: " << KVLOG(name, val, unit));
-  }
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
If there's a `recordAtomic()` why to lock her?

Reply:
Yes, removed recordAtomic() function. As discussed record will fill the values in the histogram. Boost histogram uses thread safe operator().